### DISCUSSION TITLE
[Feedback] [UI] Poor accessibility of posts with many comments

### DISCUSSION THREAD

**Discussion by MasterInQuestion** [community/community#39463] [Discussions]
Current web UI cannot adequately scale to handle large posts (e.g. 1,000,000 comments).
    The single-page (one-page) design provides poor accessibility: failing at hundreds.

    Live examples:
    [ https://github.com/MasterInQuestion/TEST/issues/4#issuecomment-1314215587](https://github.com/MasterInQuestion/TEST/issues/4#issuecomment-1314215587)
    https://github.com/orgs/community/discussions/23204
    [ https://github.com/mozilla-mobile/fenix/issues/17373#issuecomment-932700252](https://github.com/mozilla-mobile/fenix/issues/17373#issuecomment-932700252)

    See also:
    https://github.com/orgs/community/discussions/139005#discussioncomment-10826647
    https://github.com/orgs/community/discussions/139935#discussioncomment-11372528

    [ Quote jcready @ CE 2024-10-25 00:04:09 UTC:
[https://github.com/orgs/community/discussions/139935#discussioncomment-11046492](https://github.com/orgs/community/discussions/139935#discussioncomment-11046492)
    I unsubscribed from this issue.
    I got a notification that I was mentioned: I open it and I can't find where I was most recently mentioned.
    I tried sorting by "New", but this just sorts by the parent thread: completely ignoring the replies.
    I see a "Load more" thing in the middle of all the comments. I click it:
    It loads like 10 more comment of the 90+ hidden comments.
    And now the "Load more" thing is gone. ]

**Comment by benjiallen** [comment]
@MasterInQuestion thanks for the feedback, can you help understand if this is a general access issue or an issue that disproportionately affects people with disabilities?

**Reply by ghostinhershell** [reply]
Hey @MasterInQuestion, 

I wanted to ping you here again just in case you missed @benjiallen's question above. Any additional clarification and feedback would help us point you in the right direction. Thanks!

**Reply by MasterInQuestion** [reply]
Thanks for your attention.
    This is a general access issue: just try seeking to any particular comment in the presented examples: note its awkwardness.

    As whether this would disproportionately affect people with disabilities:
    I'm not quite sure. And I'm not familiar with accessibility support.

**Comment by SantjagoCorkez** [comment]
My 2 cents into this.

If a page has large amount of comments, the page only shows some amount of the newest ones, prepending the list of comments with a "Load earlier comments..." button.

That button itself only loads a limited amount of comments, so if the actual number of comments is really large, only a portion of older comments gets loaded and the user needs to press the button again. And again. And again... And at some point the user gets banned because of HTTP/429 Rate Limit.

I've tested that without being authenticated. And after I've authenticated I've tried once more and seems authenticated users are not rate-limited.

But still are required to spam the "Load earlier comments..." button over and over again.

So, if restriction on loading the whole load of the comments is more or less understandable if requested by an anonymous user (to prevent denials of service and/or heavy loads on the backend infrastructure), the same behavior while the user is authenticated cannot be understood. If an authenticated user doesn't get rate-limited whatever happens, why such a user cannot just optionally choose to load the whole bulk at once?

I've just got my browser accidentally crash and when I re-opened the browser the page reverted to its initial state so I needed to spam the button again just to get the whole conversation thread.

**Reply by MasterInQuestion** [reply]
One-page wouldn't reliably work whatsoever.
    The proper solution is pagination.

**Reply by adamency** [reply]
@MasterInQuestion Pagination is the worst solution to suggest to Github, especially since they never allow you to choose the number of elements per page. Have you never tried to search for a specific commit, release, package, or tag of a project in github ? You literally take dozens of minutes clicking hundreds of times on that same button, until finding the page which holds the one you were looking for. While with a git repository, it is found in 5 seconds.

No, the proper solution is to be able to **search** the full content. And having all comments on the same page is actually one of the few good things that Github has done, since we can thus have a global view of the evolution of the issue from the start to now.

**Reply by SantjagoCorkez** [reply]
I'd vote for an option to load the whole set of comments at once being available to at least authenticated users. Another separate button near "Load earlier comments..." or a checkbox (unchecked by default) which switches that "Load earlier comments..." button's behavior. That'd at least improve the usability since would provide one with an option to download everything with a single click on once's choice. Would be of more balance between default lazy load which of course is designed in favor of lowering the backend load and an option to still override in some (I presume) rare cases when that's needed.

**Reply by MasterInQuestion** [reply]
[ Quote adamency @ CE 2023-02-24 14:33:46 UTC:
[ https://github.com/orgs/community/discussions/39463#discussioncomment-5100139](https://github.com/orgs/community/discussions/39463#discussioncomment-5100139)
    Pagination is the worst solution to suggest to Github, especially since they never allow you to choose the number of elements per page. ]
<^>    While pagination is a must.
    The latter is a different problem.

    “Have you never tried to search for a specific commit, release, package, or tag of a project in github?
    You literally take dozens of minutes clicking hundreds of times on that same button, until finding the page which holds the one you were looking for.”
<^>    I don't. Request parameters exist to request navigation to the desired page. [ Verified "commits", "releases". ]
    But I admit the "commits" is much user unfriendly, comparing to the revision history page in MediaWiki.
    E.g. [ https://en.wikipedia.org/?action=history](https://en.wikipedia.org/?action=history)

[ "Search commits"
    See the example links on my profile page:
    [ https://github.com/MasterInQuestion](https://github.com/MasterInQuestion)

    Also:
    https://github.com/search?q=repo:curl/curl+DoH&type=commits&s=committer-date&o=desc
    https://github.com/search?q=repo:MasterInQuestion/attach+committer-date:>=2024-03-17&type=commits&s=committer-date&o=asc ]
\
\
    "No, the proper solution is to be able to search the full content."
<^>    This is desirable. But irrelevant.

    "And having all comments on the same page is actually one of the few good things that Github has done, ..."
<^>    The scheme is unsustainable. (resource depleting)
    One example: https://github.com/microsoft/vscode/milestone/8

**Comment by jglick** [comment]
Many of GitHub’s own discussion pages quickly become unreadable. I left a comment somewhere in https://github.com/orgs/community/discussions/143787 and now

> You’re receiving notifications because you commented.

which is OK but if the original poster actually replied to _me_ in the middle of dozens of other threads I would be hard-pressed to find that out. Searching for my user name via browser Ctrl-F does not work because my thread is long since hidden. I can PageDown for a while until I see

> 67 hidden items
> **Load more…**

but clicking that still does not show mine, and instead I need to search again for

> 37 hidden items
> **Load more…**

then

> 7 hidden items
> **Load more…**

and finally I can search for my user name and find my comment, only to see that there has been no response yet. This is terribly inconvenient. **Issues** would have been a better format.

**Reply by MasterInQuestion** [reply]
For different intent we favor different view:
    https://github.com/orgs/community/discussions/139935#discussioncomment-11058053
    (quick trace vs. proper context)

    Note also:
    https://github.com/orgs/community/discussions/139005#discussioncomment-11142315
    https://github.com/orgs/community/discussions/139005#discussioncomment-10892671
    “The primary problem is the bad segmentation:
    Things of high relevance scattered far away each.”
    .
    May worth reference:
    https://trac.ffmpeg.org/ticket/11055#comment:2
    https://github.com/orgs/community/discussions/142308#discussioncomment-11974505
    ; refactoring maybe required to aid readability.
    (where simple timeline would fail)

    [ nielsdos @ CE 2025-01-26 18:33:25 UTC:
[https://github.com/orgs/community/discussions/148713#discussioncomment-11960477](https://github.com/orgs/community/discussions/148713#discussioncomment-11960477)
    Not showing the entire discussion upon loading an issue page, and having to click "Load more" makes no sense to me.
    If I click on an issue I want to see the latest discussion that happened. ]

**Comment by MasterInQuestion** [comment]
"Show the whole conversation in Discussions. It's better UX and simpler code."
    Out-Of-Memory for really big threads would be another problem.
[[
<b id="discussioncomment-12095762">[ verdverm @ CE 2025-02-07 15:38:59 UTC:
[https://github.com/orgs/community/discussions/148713#discussioncomment-12095762](https://github.com/orgs/community/discussions/148713#discussioncomment-12095762)</b>
    If OOM is really an issue for Discussions, there are more serious problems at Microsoft GitHub than I realized.
    There are plenty of other sites that can load far more, than a long threaded discussion would ever come close to. <b>]</b>

\----

<b id="discussioncomment-12206195">[ MasterInQuestion @ CE 2025-02-14 22:46:18 UTC:
[https://github.com/orgs/community/discussions/148713#discussioncomment-12206195](https://github.com/orgs/community/discussions/148713#discussioncomment-12206195)</b>
    E.g. [https://github.com/orgs/community/discussions/148713#discussioncomment-11864333](https://github.com/orgs/community/discussions/148713#discussioncomment-11864333)
    ; this link wouldn't normally load due to the glitched janky anchor navigation.
    (accessible only after expanded the relevant collapsed)

    Current browsers cannot well-handle extraordinarily lengthy content.
    Refer also: https://github.com/orgs/community/discussions/139005#discussioncomment-10817089 <b>]</b>

---- (1)

<b id="discussioncomment-10817089">[ MasterInQuestion @ CE 2024-10-02 07:46:11 UTC:
[https://github.com/orgs/community/discussions/139005#discussioncomment-10817089](https://github.com/orgs/community/discussions/139005#discussioncomment-10817089)</b>
    "It must be taken in account, there will be always a diff too big to display for the browser..."

    By proper segmentation and dynamic loading, sort of like what mentioned in:
    [ https://github.com/MasterInQuestion/attach/commit/401c58fc3a56d265feadda567f02d79de93ff10c#r140161715](https://github.com/MasterInQuestion/attach/commit/401c58fc3a56d265feadda567f02d79de93ff10c#r140161715)
    ; big files shouldn't really be the problem.

    Ideally the browser should self-handle alike:
    https://news.ycombinator.com/item?id=40015961#40026937 <b>]</b>
]]
